src/hockeypuck/openpgp/io.go

```diff
@@ -53,8 +84,12 @@ func WritePackets(w io.Writer, key *PrimaryKey) error {
 	return nil
 }
 
-func WriteArmoredPackets(w io.Writer, roots []*PrimaryKey) error {
-	armw, err := armor.Encode(w, openpgp.PublicKeyType, nil)
+func WriteArmoredPackets(w io.Writer, roots []*PrimaryKey, options ...KeyWriterOption) error {
+	akwr, err := NewArmoredKeyWriter(options...)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	armw, err := armor.Encode(w, openpgp.PublicKeyType, akwr.headers)
 	defer armw.Close()
```

Comment:
It looks like armor.Encode can return a nil writer if err !=nil (see https://github.com/golang/crypto/blob/0c34fe9e7dc2/openpgp/armor/encode.go#L133), so it'd be good to move this below the error check.